docs/user_guide/fairness_in_machine_learning.rst

```diff
@@ -141,6 +141,94 @@ which groups these values were observed, as well as the difference and ratio
+process. For example, using risk assessment to decide whether a person is eligible
+for mortgage approval. Another example is using risk assessment to decide whether a
+person is likely to re-engage in criminal behaviour as they move through the criminal
+justice pipeline.
+
+As far as the outcomes of these decisions go, in the first scenario an individual may be
+approved or not to get a mortgage which then impacts their ability to have housing security,
+gain wealth and social status. In the second scenario an individual may be released pretrial,
+or bail may be required. This assessment may also include sentencing and parole hearing.
+Risk assessment tools may be used as recommendations towards an outcome. However there is no
+guarantee that they will be used consistently and thus bias is introduced. For example, a
+party responsible for choosing the outcome decides to ignore the recommendation as a result
+of personal bias, even when the model takes into account demographic data. In this example
+a model that fails to take into account bias present in those judging culminates in a
+biased result.
+
+The Portability Trap
+^^^^^^^^^^^^^^^^^^^^
+
+Selbst et al. [#4]_ define this as a "failure to understand how repurposing algorithmic
+solutions designed for one social context may be misleading, inaccurate, or otherwise
+do harm when applied to a different context".
+
+It is the practice of reusing others work without taking into account lack of
+transferability to a given social situation, resulting in harmful consequences.
+
+The Formalism Trap
+^^^^^^^^^^^^^^^^^^
+
+Selbst et al. [#4]_ define this as a "failure to account for the full meaning of social concepts
+such as fairness, which can be procedural, contextual, and contestable, and cannot be resolved
+through mathematical formalisms".
+
+It is the practice of implementing mathematical and statistical models, along with
+corresponding assumptions, that fail to take into consideration the social, demographic,
+economic, or otherwise non-technical aspect that make up the phenomenon being studied.
+
+The Ripple Effect Trap
+^^^^^^^^^^^^^^^^^^^^^^
+
+Selbst et al. [#4]_ define this as a "failure to understand how the insertion of technology into
+an existing social system changes the behaviors and embedded values of the pre-existing system".
+
+
+The Solutionism Trap
+^^^^^^^^^^^^^^^^^^^^
+
+Selbst et al. [#4]_ define this as a "failure to recognize the possibility that the best solution
+to a problem may not involve technology".
+
+It is the practice of assuming that a machine learning algorithm is the best solution to a problem.
+
+One area where this manifests in contexts in which the definition "fairness" changes or is dependent
+on a political context. When this happens, models become obsolete. Another area where this manifests
+is when the question at hand requires a computationally intractable solution.
```

Comment:
An example would help make this piece clearer as well

Reply:
added example